api/v1alpha1/user_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// UserSpec defines the desired state of S3User
+type UserSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Name of the S3User
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Password of the S3User
+	// +kubebuilder:validation:Optional
+	Password string `json:"password,omitempty"`
```

Comment:
Password should go in a referenced secret.

You can use [`v1.SecretKeySelector`](https://pkg.go.dev/k8s.io/api@v0.29.3/core/v1#SecretKeySelector) for this.

Reply:
password is optional if u think you will not use this property, i think we should go with no password available in the CR in order to minimize the work

Reply:
username should be referenced as secretKey